### PR TITLE
Add SetChannelFlushStatus to Endpoint

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

Similar to SetChannelStatus, replaces the typical get channel/mutate/set channel sequence.

closes: #3947 

### Commit Message / Changelog Entry

```bash
chore: Add SetChannelFlushStatus to Endpoint
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by DimitrisJim** [comment]
can also replace usages currently used.

File: testing/endpoint.go [hunk 1]
```diff
@@ -737,6 +737,18 @@ func (endpoint *Endpoint) SetChannelState(state channeltypes.State) error {
 	return endpoint.Counterparty.UpdateClient()
 }
 
+// SetChannelFlushStatus sets the channel flush status
+func (endpoint *Endpoint) SetChannelFlushStatus(status channeltypes.FlushStatus) error {
```
Comment (charleenfei) [inline testing/endpoint.go:741]:
```suggestion
func (endpoint *Endpoint) SetChannelFlushStatus(status channeltypes.FlushStatus) {
```

File: testing/endpoint.go [hunk 2]
```diff
@@ -730,7 +730,19 @@ func (endpoint *Endpoint) SetChannelState(state channeltypes.State) error {
 	channel := endpoint.GetChannel()
 
 	channel.State = state
-	endpoint.Chain.App.GetIBCKeeper().ChannelKeeper.SetChannel(endpoint.Chain.GetContext(), endpoint.ChannelConfig.PortID, endpoint.ChannelID, channel)
+	endpoint.SetChannel(channel)
```
Comment (crodriguezvega) [inline testing/endpoint.go:733]: @DimitrisJim I noticed that we could use the helper function here as well, so I pushed the change in your branch. I hope that's ok!

**Comment by DimitrisJim** [comment]
Going to close this in preference to what Cian describes. The functionality can be added in one piece after 04-channel-upgrades has been merged so as to not hold it back. Some more context here: https://github.com/cosmos/ibc-go/issues/3955#issuecomment-1610919440

**Pull request closed without merging**